deepdives/build-process/README.md

```diff
@@ -0,0 +1,66 @@
+
+###### Readings
+- [Docs](http://browserify.org/)
+- [Browserify Handbook](https://github.com/substack/browserify-handbook)
+- [Getting Started with Browserify](http://www.sitepoint.com/getting-started-browserify/)
+
+##### Webpack
+![Webpack](https://camo.githubusercontent.com/f1d103872f836f33dbff7a74ed819004f792a4ad/687474703a2f2f7765627061636b2e6769746875622e696f2f6173736574732f6c6f676f2e706e67)
+
+Webpack is the current darling of the frontend programming world. It is an incredibly powerful bundling tool, but initally groking (understanding) it can be difficult. Weback can be configured to deal with any file type by using a variety of ```loaders``` which teach webpack how to handle each type of file. These loaders can be very powerful such as the babel loader which will compile your javascript files before bundling them. It can use any module system including both node like syntax or the es6 module system. Webpack is incredibly versatile and powerful and can also do most of the things you would need a task runner (see below) to do. It if frequently used entirely on its own.
+
+###### Readings
+- [Docs](http://webpack.github.io/docs/)
+- [Beginner's Guide to Webpack](https://medium.com/@dabit3/beginner-s-guide-to-webpack-b1f1a3638460#.pww8hq4f2)
+- [Webpack Howto](https://github.com/petehunt/webpack-howto#webpack-howto)
+
+### Task Runners
+If you want to coordinate your various task, such as compling code, checking for syntax errors, running test, bundling your javascript. You may want to use a task runner. They make it easy to configure and automate running tasks for you. The two most common ones are grunt and gulp.
+
+##### Grunt
+![Grunt](http://www.chloechen.io/wp-content/uploads/2014/11/logo.png)
+
+Grunt works by creating a grunfile which defines a configuration object with a list of tasks and then registering them to commands or events. It is commonly called "configuration focused"
+
+###### Readings
+- [Docs](http://gruntjs.com/)
+- [Best Practices for Building Angular-js Apps](https://medium.com/@dickeyxxx/best-practices-for-building-angular-js-apps-266c1a4a6917#.1ridn33pb)
+- [Grunt for People Who Think Things Like Grunt are Weird and Hard](https://24ways.org/2013/grunt-is-not-weird-and-hard/)
+
+##### Gulp
+![Gulp](http://www.geekhive.com/~/media/Images/Buzz/Post%20Images/2014/9/gulp.ashx)
+
+Gulp works by piping files through a series of specialized plugins. It uses straight javascript mostly  node streams and promises. It can be a bit harder to wrap your head around but generally the files are shorter and they run faster.
```

Comment:
In contrast to Grunt, Gulp pipes files through a series of specialized plugins. It relies on some JavaScript features you might be familiar with -- namely, node streams and promises. Gulp can be a bit harder to wrap your head around, but generally the config files are shorter and run faster.